### PR TITLE
Allow for forceing the update method

### PR DESCRIPTION
**Pull request by begizi**
even if the boundaries haven't changed.

Useful when registering an update event that doesn't modify the
boundaries at all. ie. filtering the original collection.

### CODE REVIEW

File: src/marionette.sliding-view.js [hunk 1]
```diff
@@ -76,7 +81,7 @@ Mn.SlidingView = Mn.CollectionView.extend({
     }
 
     // If the boundaries are unchanged, then we bail out early
-    if (lowerBound === this._lowerBound && upperBound === this._upperBound) {
+    if (!this.forceUpdate && (lowerBound === this._lowerBound && upperBound === this._upperBound)) {
```
Comment (jmeas) [inline src/marionette.sliding-view.js]: This conditional has gotten a little out of hand. Maybe...

```js
var boundsUpdated = lowerBound !== this._lowerBound && upperBound !== this._upperBound;

if (!this.forceUpdate && !boundsUpdated) {
  // toop toop
}
```

also, could you update the comment?

**Comment by jmeas** [comment]
I'm curious to know more about the use case here.

> Useful when registering an update event that doesn't modify the
> boundaries at all. ie. filtering the original collection.

Do you mean that there are edge cases where filtering doesn't change the boundaries? I would expect filtering to mostly update which things are visible :open_mouth: 

More importantly, though, do you plan to be filtering at 60fps? This library is pretty much intended for 60fps ops, like scrolling. I'm hesitant to add something like filtering to this, as it seems like functionality that would be easy enough to add outside of this whole throttling system.

File: src/marionette.sliding-view.js (hunk 1 above)
Comment (begizi) [inline src/marionette.sliding-view.js]: Updated.

File: src/marionette.sliding-view.js [hunk 2]
```diff
@@ -75,8 +80,9 @@ Mn.SlidingView = Mn.CollectionView.extend({
       }
     }
 
-    // If the boundaries are unchanged, then we bail out early
-    if (lowerBound === this._lowerBound && upperBound === this._upperBound) {
+    // If the boundaries are unchanged and we're not forcing updates then abort the update
+    var boundsUpdated = lowerBound === this._lowerBound && upperBound === this._upperBound;
```
Comment (jmeas) [inline src/marionette.sliding-view.js]: This variable name doesn't match what it's checking. If it's `true`, then the bounds *haven't* updated.

**Comment by begizi** [comment]
Due to the library handling it's own collection view under the hood, it's not handling the standard collection events until the `_updateCollection` method gets called.

I think a more robust solution to this issue may be to make `_updateCollection` a public method (`updateCollection`)

What do you think about cloning the referenceCollection including it's event listeners into the views collection?

**Comment by jmeas** [comment]
> I think a more robust solution to this issue may be to make _updateCollection a public method (updateCollection)

:+1: to that.

> What do you think about cloning the referenceCollection including it's event listeners into the views collection?

I think this is a fine idea to implement in your own extension of this library. I'm worried about adding too many hooks into this library to update the boundaries and then trigger the update event as things get added/removed from the collection.

Like Backbone itself, this library isn't meant to be a handles-all-situations-out-of-the-box kinda library. It's lightweight, and provides one with the most minimal pattern (cloning the collection) and fewest hooks (a throttled callback) one needs to manage 60fps ops, which is otherwise a difficult thing to do.

*With that said*, if you're clever enough to come up with a small API that handles updating after the built-in Collection events, I'd be open to merging it.

**Comment by begizi** [comment]
I'll open a new PR with the private to public change. I agree it's a better solution than this force option.

**Pull request closed without merging**